frontend/util/pathutil.ts

```
@@ -0,0 +1,291 @@
+ */
+export function hasInvalidChars(path: string, platform: string): boolean {
+    if (platform === PlatformWindows) {
+        // Windows reserved characters (except \ and / which are path separators)
+        // < > : " | ? * and control characters
+        // Note: : is allowed as second char for drive letter
+        const pathWithoutDrive = path.length >= 2 && path[1] === ":" ? path.substring(2) : path;
+        if (/[<>"|?*]/.test(pathWithoutDrive)) {
+            return true;
+        }
+        // Control characters (0x00-0x1F)
+        if (/[\x00-\x1F]/.test(path)) {
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)